app/src/androidTest/java/ch/epfl/sdp/blindly/fake_module/FakeAppDatabaseModule.kt

```diff
@@ -0,0 +1,90 @@
+package ch.epfl.sdp.blindly.fake_module
+
+import android.content.Context
+import androidx.room.Room
+import ch.epfl.sdp.blindly.database.localDB.AppDatabase
+import ch.epfl.sdp.blindly.database.localDB.UserDAO
+import ch.epfl.sdp.blindly.database.localDB.UserEntity
+import dagger.Module
+import dagger.hilt.components.SingletonComponent
+import dagger.hilt.testing.TestInstallIn
+import ch.epfl.sdp.blindly.dependency_injection.AppDatabaseModule
+import ch.epfl.sdp.blindly.location.AndroidLocationService
+import ch.epfl.sdp.blindly.user.User
+import dagger.Provides
+import dagger.hilt.android.qualifiers.ApplicationContext
+import org.mockito.Mockito
+import javax.inject.Singleton
+
+
+@Module
+@TestInstallIn(
+    components = [SingletonComponent::class],
+    replaces = [AppDatabaseModule::class]
+)
+
+open class FakeAppDatabaseModule {
+    companion object {
+        private const val uid = "abcd1234"
+        private const val username = "Jane Doe"
+        private val location =
+            AndroidLocationService.createLocationTableEPFL() // Ecublens, Switzerland
+        private const val birthday = "01.01.01"
+        private const val gender = "Woman"
+        private val sexualOrientations = listOf("Asexual")
+        private const val showMe = "Everyone"
+        private val passions = listOf("Coffee", "Tea")
+        private const val radius = 150
+        private val matches: List<String> = listOf("a1", "b2")
+        private val likes: List<String> = listOf("c3", "d4")
+        private const val recordingPath =
+            "Recordings/OKj1UxZao3hIVtma95gWZlner9p1-PresentationAudio.amr"
+        private val ageRange = listOf(30, 50)
+        val fakeUser = User.Builder()
```

Comment:
Maybe you could use the fakeuser that is already in the fakeUserCache since it's a duplicate